calcom/cal.com
# Enable apps for managed event types & other team event types

## What does this PR do?



This PR does the following:
- Enable apps for managed event types
- Share credentials for team event types without having the event type owner present in the booking

*I did not make changes to Rainbow as it is now depreciated*

For apps like ga4 & Fathom, there are no access tokens that are needed; we can assume any team admins can enable these apps for event types types. Although for Stripe since an access token is needed we use the team owner's credentials for all event types.

### Follow up
- For video apps, all team members still need to have the app installed. I think we should give the option of using a single video credential for all bookings created under the team event type or allow team members to set it themselves 
- When handling Stripe credentials a lot of them are hard coded for Stripe. Should we handle payment credentials under the `PaymentService`?
- Tests should be made under each app folder to ensure that this behaviour is acting as expected

Fixes # (issue)



**Environment**: Staging(main branch) / Production

## Type of change



- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)


## How should this be tested?



- See enabled apps on team event types (not including Stripe)
  - Create a team and invite another admin
  - Create a managed event type
  - Install and enable some or all of the event type apps except Stripe & Rainbow
  - Now, visit that event type as the other team admin. Even if the admin does not have the apps installed, they should still see them enabled on the team event type.
  - Still, as the admin go to one of the user's event types. The uninstalled apps should not appear on the event type apps page.
  - In the managed event type, assign the event type to a team member
  - Under the child event type go to the apps tab. The apps should not be editable.

- Enable Stripe for team events
    - Create any team event as the team owner
    - Install Stripe
    - Enable Stripe and set the payment option to collect money on booking 
    - Assign the event type to members that are not the owner
    - Create a booking. The payment should go through without the team owner present in the booking
    - Repeat the above steps for no-show fee bookings
    - Visit the event type as a team admin. You should not be able to edit any Stripe fields
    - As the team owner, uninstall Stripe. It should disappear from any team events

## Checklist



- I haven't added tests that prove my fix is effective or that my feature works
- I haven't checked if new and existing unit tests pass locally with my changes
